cmake/CMakeLists.txt

```diff
@@ -120,6 +120,7 @@ option(onnxruntime_DISABLE_RTTI "Disable RTTI" OFF)
 # For now onnxruntime_DISABLE_EXCEPTIONS will only work with onnxruntime_MINIMAL_BUILD, more changes (ONNX, non-CPU EP, ...) are required to run this standalone
 option(onnxruntime_DISABLE_EXCEPTIONS "Disable exception handling. Requires onnxruntime_MINIMAL_BUILD currently." OFF)
 option(onnxruntime_MINIMAL_BUILD "Exclude as much as possible from the build. Support ORT format models. No support for ONNX format models." OFF)
+option(onnxruntime_EXTENDED_MINIMAL_BUILD "onnxruntime_MINIMAL_BUILD with support for execution providers that compile kernels." OFF)
```

Comment:
Would the scope of extended minimal build expand under the same option name? If not, suggest naming this specific to the functionality it is enabling.

Reply:
Not totally clear. I was thinking it we were to add support for runtime optimization there would be some overlap and both could be considered 'extended' functionality. I'm also trying to keep the number of different #ifdef values in the code under control. For example if we added runtime optimization you're getting into something horrendous for Graph::AddNode of `#if !defined(ORT_MINIMAL_BUILD) || defined(ORT_COMPILING_EP_MINIMAL_BUILD) || defined(ORT_RUNTIME_OPT_MINIMAL_BUILD)`. And add one more feature that overlaps and it gets worse.

So I went with something generic we can add to until we have a hard requirement to split these additional functionalities out separately.

Reply:
Yeah, this is fine. More functionality will be covered under 'extended'. You answered my question.